tests/unit/Infrastructure/DashboardBundle/Factory/EntityTypeFactoryTest.php

```diff
@@ -225,4 +226,140 @@ public function test_build_create_new_oauth20_rs_form()
 
         $this->assertInstanceOf(FormType::class, $form);
     }
+
+    /**
+     *  @dataProvider testOptions
+     */
+    public function test_build_edit_saml_form_from_entity(
+        $isCopy,
+        $isRequestedProductionEntity,
+        $manageEntityStatus,
+        $environment,
+        $expectation
+    ) {
+        $manageEntity = m::mock(ManageEntity::class);
+        $manageEntity
+            ->shouldReceive('isRequestedProductionEntity')
+            ->with($isCopy)
+            ->andReturn($isRequestedProductionEntity);
```

Comment:
Might be out of scope for this test, but you could consider making the `ManageEntity` a partial mock. You could make the `isRequestedProductionEntity` method the only 'real' method  that is executed (and is not mocked). That way you can make your data provider simpler. As you no longer have to provide the `isRequestedProducitonEntity` and `manageEntityStatus`.

If you'd like me to demonstrate this feature, feel free to hook up IRL.